src/lreg.jl

```diff
@@ -20,8 +20,13 @@ _haug(X::AbstractMatrix{T}) where T = hcat(X, ones(T, size(X,1), 1))::Matrix{T}
 ## linear least square
 
 function llsq(X::AbstractMatrix{T}, Y::AbstractVecOrMat{T};
-              trans::Bool=false, bias::Bool=true) where {T<:Real}
-    if trans
+              trans::Bool=false, bias::Bool=true,
+              dims::Union{Integer,Nothing}=nothing) where {T<:Real}
+    if dims === nothing
+        Base.depwarn("`trans` argument is deprecated, use llsq(X, Y, dims=d) instead.", :trans)
```

Comment:
AFAICT this warning will be printed even if the caller didn't set `trans`, in which case it will be confusing. I guess it should say that `dims` is required, and only mention `trans` if it's `true`, right?

Also, the default for `dims` should be `1` until we remove the deprecation, or that will break code.